ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -57,19 +66,142 @@ private[apiserver] final class ApiUserManagementService(
+            metadata = ObjectMeta(
+              resourceVersionO = None,
+              annotations = pAnnotations,
+            ),
+          ),
+          pRights,
+        )
       } { case (user, pRights) =>
         userManagementStore
           .createUser(
             user = user,
             rights = pRights,
           )
           .flatMap(handleResult("creating user"))
-          .map(_ => CreateUserResponse(Some(request.user.get)))
+          .map(createdUser => CreateUserResponse(Some(toProtoUser(createdUser))))
+      }
+    }
+
+  override def updateUser(request: UpdateUserRequest): Future[UpdateUserResponse] = {
+    withSubmissionId { implicit loggingContext =>
+      // Retrieving the authenticated user from the context
+      val authorizedUserIdFO: Future[Option[String]] = resolveAuthenticatedUser()
+      withValidation {
+        for {
+          pUser <- requirePresence(request.user, "user")
+          pUserId <- requireUserId(pUser.id, "user.id")
+          pMetadata = pUser.metadata.getOrElse(
+            com.daml.ledger.api.v1.admin.object_meta.ObjectMeta()
+          )
+          pFieldMask <- requirePresence(request.updateMask, "update_mask")
+          pOptPrimaryParty <- optionalString(pUser.primaryParty)(requireParty)
+          pResourceVersion <- optionalString(pMetadata.resourceVersion)(
+            FieldValidations.requireResourceVersion(_, "user.metadata.resource_version")
+          )
+          pAnnotations <- verifyMetadataAnnotations(
+            pMetadata.annotations,
+            "user.metadata.annotations",
+          )
+        } yield (
+          User(
+            id = pUserId,
+            primaryParty = pOptPrimaryParty,
+            isDeactivated = pUser.isDeactivated,
+            metadata = ObjectMeta(
+              resourceVersionO = pResourceVersion,
+              annotations = pAnnotations,
+            ),
+          ),
+          pFieldMask,
+        )
+      } { case (user, fieldMask) =>
+        for {
+          userUpdate <- handleUpdatePathResult(user.id, UserUpdateMapper.toUpdate(user, fieldMask))
+          _ <-
+            if (userUpdate.isNoUpdate) {
+              Future.failed(
+                LedgerApiErrors.Admin.UserManagement.InvalidUpdateUserRequest
+                  .Reject(
+                    userId = user.id,
+                    reason = "Update request describes a no-up update",
+                  )
+                  .asGrpcError
+              )
+            } else {
+              Future.successful(())
+            }
+          authorizedUserIdO <- authorizedUserIdFO
+          _ <-
+            if (
+              authorizedUserIdO
+                .contains(userUpdate.id) && userUpdate.isDeactivatedUpdateO.contains(true)
+            ) {
+              Future.failed(
+                LedgerApiErrors.RequestValidation.InvalidArgument
+                  .Reject(
+                    "Requesting user cannot deactivate itself"
```

Comment:
```suggestion
                    "Requesting user cannot self-deactivate"
```
Pronoun "it" probably doesn't apply to an animate user.

Reply:
Ok